### ISSUE TITLE
kubelet logs "conversion.go:128 failed to handle multiple devices for container. Skipping Filesystem stats"

### ISSUE THREAD

**Issue by justinsb** [kubernetes/kubernetes#22586]
Not sure exactly what this means, but recording as a TODO.

We do use a symlink on AWS from /var/lib/docker (and also for the kubernetes dir I think) to /mnt/ephemeral. If this isn't happening on GCE, probably related to that!

**Comment by justinsb** [comment]
Marking as a P3.  I suspect it is actually more important, but we can upgrade once I figure out what the consequence is.

**Comment by colemickens** [comment]
I'm seeing this in clusters when I mount the Azure ephemeral drive to /var/lib/docker with a systemd mount unit:

https://github.com/colemickens/kubernetes-anywhere/blob/87cbb20881114e843cca8d266c0930544f2abb30/phase2/ansible/playbooks/roles/node/templates/azure/var-lib-docker.mount

Did you ever resolve it by chance?

**Comment by colemickens** [comment]
I can confirm I'm seeing this even when /var/lib/docker is completely unmodified and is just a normal directory on the root drive. I'm at a loss at to what is causing it or how to fix it.

**Comment by ScubaDrew** [comment]
Seeing this on my 1.2.3 AWS k8s-1.2-debian-jessie-amd64-hvm-2016-03-05-ebs (ami-7840ac18) node.

Drew

**Comment by cristifalcas** [comment]
We get this on a cluster build with Vmware machines. We use devicemapper with xfs.

**Comment by gg7** [comment]
Seeing this as well. Running Kubernetes 1.4.0, bare-metal Ubuntu 16.04 (x86_64), Docker 1.11.2 (Storage Driver: aufs, Backing Filesystem: zfs).

**Comment by MaikuMori** [comment]
Seeing this on GKE.

**Comment by jjungnickel** [comment]
Kubernetes 1.4.3, CoreOS 1068.8.0 on vSphere reporting in. imagefs is ext4 on /var/lib/docker

**Comment by spxtr** [comment]
Seeing this on GKE with 1.4.5 nodes. It's definitely not AWS, but I'm not sure what team to point it at, and I'm also not sure what priority it deserves.

**Comment by atomantic** [comment]
This is logging pretty heavily in GKE to stackdriver, contributing heavily to 22MB/minute of GKE logs from my cluster into stackdriver.

**Comment by MikeSpreitzer** [comment]
I am also seeing this, with line number 133, from v1.4.4.  I am running on "bare metal" (really VirtualBox VMs, but no cloud provider in sight) Ubuntu 16.04, running the kubelet inside a Docker container.  Here's the relevant line from my systemd unit on a master machine (the simpler case, and I get the error message only twice):
```
ExecStart=/usr/bin/docker run \
		--name=kubelet \
		--privileged \
		--userns=host \
		--network=host \
		--pid=host \
		-v /:/rootfs:ro \
		-v /sys:/sys:ro \
		-v /dev:/dev \
		-v /var/run:/var/run \
		-v /var/lib/kubelet:/var/lib/kubelet \
		-v /var/lib/docker:/var/lib/docker \
		-v /etc/kubernetes:/etc/kubernetes \
		gcr.io/google_containers/hyperkube-amd64:v1.4.4 /hyperkube kubelet \
		--allow-privileged=true \
		--config=/etc/kubernetes/manifests \
		--v=2 \
		--file-check-frequency=5s \
		--hostname-override="100.127.0.2" \
		--pod-infra-container-image="gcr.io/google_containers/pause:latest"
```

**Comment by MikeSpreitzer** [comment]
CC @huang195

**Comment by dashpole** [comment]
This occurs on kubernetes head, using OS images: CoreOS-stable, GCI.  It does not occur on container-vm or ubuntu-trusty.

**Comment by dashpole** [comment]
The coreos and GCI images use multiple devices (dev/root, dev/sda1, dev/sda8, for example).  Whereas container-vm and ubuntu-trusty only use one filesystem device (dev/sda1).

**Comment by dashpole** [comment]
This error occurs when running an image with multiple devices.  All devices are assigned to the "root" filesystem, and cadvisor doesn't know how to handle multiple devies.

**Comment by dashpole** [comment]
cc: @derekwaynecarr

**Comment by dashpole** [comment]
Once cAdvisor is vendored into k8s, this should no longer be an issue.  If people still see this problem after the next release, feel free to reopen this issue.  For now I propose closing it.

**Comment by dashpole** [comment]
This issue can be closed, as #40095 is merged
